packages/java/src/node/java-util.ts

```diff
@@ -0,0 +1,81 @@
+/********************************************************************************
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { NodePackage, ExtensionPackage } from '@theia/application-package';
+import { ApplicationPackageManager } from '@theia/application-manager';
+import { injectable } from 'inversify';
+import * as paths from 'path';
+
+/**
+ * Represents contributes section from package.json.
+ * Contributes section describes paths to the java bundles and looks like:
+ * "contributes": {
+ *   "javaExtensions": ["./server/my.java.plugin.jar"],
+ * }
+ */
+interface JavaContributes extends NodePackage {
+    /**
+     * An array of paths to the bundles.
+     */
+    javaExtensions?: string[] | undefined;
+}
+
+/**
+ * The collector to find all java bundles in Theia extensions.
+ */
+@injectable()
+export class ExtensionBundlesCollector {
+    private javaExtensions: string[] | undefined;
+
+    /**
+     * Collects paths of all java bundles declared in the extensions.
+     *
+     * @return an array of paths to the bundles
+     */
+    public collectExtensionBundlesPaths(): string[] {
+        if (this.javaExtensions) {
+            return this.javaExtensions;
+        }
+
+        const projectPath = process.cwd();
+        const manager = new ApplicationPackageManager({ projectPath });
```

Comment:
@svor I am not sure about the state of this PR, but if you continue, please use `ApplicationPackage` instead and inject it as here: https://github.com/theia-ide/theia/blob/b57e11f1f9635b36efdc5a8478248d26e3a62035/packages/typescript/src/node/typescript-contribution.ts#L33-L49

`ApplicationPackageManager` brings dev time dependencies, like `webpack`. We don't have them at runtime.

Reply:
@akosyakov I'm going to continue my work on this PR, thank you!